presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexSupplier.java

```diff
@@ -136,19 +137,24 @@ private static STRtree buildRTree(LongArrayList addresses, List<List<Block>> cha
                 partition = toIntExact(INTEGER.getLong(partitionBlock, blockPosition));
             }
 
-            rtree.insert(getEnvelope(ogcGeometry, radius), new GeometryWithPosition(ogcGeometry, partition, position));
+            rtree.insert(GeometryUtils.getJtsEnvelope(ogcGeometry, radius), new GeometryWithPosition(ogcGeometry, partition, position));
```

Comment:
Looks like `ogcGeometry.isEmpty()` check above was not sufficient. What kind of geometry would cause a problem? Is it a geometry collection of empty geometries? Would you add a test that reproduces the original issue?

Reply:
I think we should not add empty envelopes to the R-Tree. These can't match anything anyway.

Reply:
I should have described more in the PR :).  The issue is that the previous method of getting an envelope from a `GeometryCollection` did not actually get the right envelope.  The test which creates a `GeometryCollection` tests this, and fails without the code change.  Other parts of the code used the correct way of getting an envelope from a `GeometryCollection` (iterating through and extending the envelope for each entry), which I extracted to `GeometryUtils`.  I presume the `ogcGeometry.isEmpty()` check will avoid the empty envelope case; if not something else is wrong!